### PR TITLE
lint: fix lint warning

### LINKED ISSUE #5622: 🐛[BUG]src\pages\user\login\components\Login\index.tsx:  Tab should be placed after handleSubmit
src\pages\user\login\components\Login\index.tsx:
Tab should be placed after handleSubmit

### PR DESCRIPTION
**Pull request by yoyo837**
fix #5622

### CODE REVIEW

File: src/pages/user/login/components/Login/index.tsx [hunk 1]
```diff
@@ -122,6 +110,18 @@ class Login extends Component<LoginProps, LoginState> {
     }
   };
 
+  public static Tab = LoginTab;
```
Comment (afc163) [inline src/pages/user/login/components/Login/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4531674)

File: src/pages/user/login/components/Login/index.tsx [hunk 2]
```diff
@@ -122,6 +110,18 @@ class Login extends Component<LoginProps, LoginState> {
...
+
+  public static Submit = LoginSubmit;
+
+  public static UserName: React.FunctionComponent<LoginItemProps>;
+
+  public static Password: React.FunctionComponent<LoginItemProps>;
+
+  public static Mobile: React.FunctionComponent<LoginItemProps>;
+
+  public static Captcha: React.FunctionComponent<LoginItemProps>;
```
Comment (afc163) [inline src/pages/user/login/components/Login/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4531674)

File: src/pages/user/login/components/Login/index.tsx [hunk 3]
```diff
@@ -122,6 +110,18 @@ class Login extends Component<LoginProps, LoginState> {
...
 
+  public static Tab = LoginTab;
+
+  public static Submit = LoginSubmit;
+
+  public static UserName: React.FunctionComponent<LoginItemProps>;
+
+  public static Password: React.FunctionComponent<LoginItemProps>;
+
+  public static Mobile: React.FunctionComponent<LoginItemProps>;
```
Comment (afc163) [inline src/pages/user/login/components/Login/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4531674)

File: src/pages/user/login/components/Login/index.tsx [hunk 4]
```diff
@@ -122,6 +110,18 @@ class Login extends Component<LoginProps, LoginState> {
     }
   };
 
+  public static Tab = LoginTab;
+
+  public static Submit = LoginSubmit;
+
+  public static UserName: React.FunctionComponent<LoginItemProps>;
```
Comment (afc163) [inline src/pages/user/login/components/Login/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4531674)

File: src/components/NoticeIcon/index.tsx [hunk 5]
```diff
@@ -135,6 +133,8 @@ export default class NoticeIcon extends Component<NoticeIconProps> {
     }
   };
 
+  public static Tab: typeof NoticeList = NoticeList;
```
Comment (afc163) [inline src/components/NoticeIcon/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4531674)

File: src/pages/user/login/components/Login/index.tsx [hunk 6]
```diff
@@ -122,6 +110,18 @@ class Login extends Component<LoginProps, LoginState> {
...
+
+  public static Password: React.FunctionComponent<LoginItemProps>;
```
Comment (afc163) [inline src/pages/user/login/components/Login/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4531674)

File: src/pages/user/login/components/Login/index.tsx [hunk 7]
```diff
@@ -122,6 +110,18 @@ class Login extends Component<LoginProps, LoginState> {
     }
   };
 
+  public static Tab = LoginTab;
+
+  public static Submit = LoginSubmit;
```
Comment (afc163) [inline src/pages/user/login/components/Login/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4531674)

**Comment by yoyo837** [comment]
disable the lint?

**Comment by afc163** [comment]
需要升一下 eslint 和 stylelint 版本了

File: .eslintrc.js [hunk 8]
```diff
@@ -4,4 +4,7 @@ module.exports = {
     ANT_DESIGN_PRO_ONLY_DO_NOT_USE_IN_YOUR_PRODUCTION: true,
     page: true,
   },
+  rules: {
+    'react/sort-comp': 'off',
```
Comment (afc163) [inline .eslintrc.js]: 这个改动多么

Comment (yoyo837) [reply .eslintrc.js]: 就 6c8260e4b8d839c538c1b6845bd6103831a968fb 里这些

Comment (yoyo837) [reply .eslintrc.js]: 不禁用就需要改，但前面已经revert了。

Comment (yoyo837) [reply .eslintrc.js]: 陈帅 让我采用禁用的方式的。

**Comment by chenshuai2144** [comment]
把 fabric 升级到 2.0 试试。eslint6+fabric@1 会有问题

File: .eslintrc.js (hunk 8 above)
Comment (afc163) [reply .eslintrc.js]: 没啥问题就改吧，关了项目里约束就弱了。

**Comment by afc163** [comment]
> 把 fabric 升级到 2.0 试试。eslint6+fabric@1 会有问题

ci 貌似是过的？

**Comment by chenshuai2144** [comment]
使用 yarn 安装 跑不起来，会报错包找不到。

**Comment by afc163** [comment]
> 使用 yarn 安装 跑不起来，会报错包找不到。

加个 yarn 安装的 ci 。

File: package.json [hunk 9]
```diff
@@ -96,7 +96,10 @@
     "cross-env": "^6.0.0",
     "cross-port-killer": "^1.1.1",
     "enzyme": "^3.9.0",
-    "eslint": "5.16.0",
+    "eslint": "^6.7.0",
+    "eslint-plugin-eslint-comments": "^3.1.2",
+    "eslint-plugin-jest": "^23.0.4",
+    "eslint-plugin-unicorn": "^13.0.0",
```
Comment (chenshuai2144) [inline package.json]: 都删掉，换成成 fabric 就好了

File: .eslintrc.js [hunk 10]
```diff
@@ -1,10 +1,9 @@
+const fabric = require('@umijs/fabric');
+
 module.exports = {
-  extends: [require.resolve('@umijs/fabric/dist/eslint')],
```
Comment (chenshuai2144) [inline .eslintrc.js]: 这个不要改，就用这个好了

**Pull request merged**